### PR TITLE
Add full_name property in User class

### PR DESCRIPTION
**Pull request by tasi788**

### CODE REVIEW

File: pyrogram/client/types/user_and_chats/user.py [hunk 1]
```diff
@@ -182,13 +184,17 @@ def _parse(client, user: types.User) -> "User" or None:
             is_support=user.support,
             first_name=user.first_name,
             last_name=user.last_name,
+            full_name=(user.first_name + " " +
+                       user.last_name if user.last_name else None),
```
[the remark this answers is not in the document]
Comment (JosXa) [reply pyrogram/client/types/user_and_chats/user.py]: I don't see a reason why .format() should be avoided (not sure about f-strings and pyro compatibility, would have to check...)
1st option LGTM.

[the remark this answers is not in the document]
Comment (tasi788) [reply pyrogram/client/types/user_and_chats/user.py]: I've change the code, please have review.

File: pyrogram/client/types/user_and_chats/user.py [hunk 2]
```diff
@@ -182,13 +184,17 @@ def _parse(client, user: types.User) -> "User" or None:
...
             username=user.username,
             language_code=user.lang_code,
             dc_id=getattr(user.photo, "dc_id", None),
             phone_number=user.phone,
-            photo=ChatPhoto._parse(client, user.photo, user.id, user.access_hash),
-            restrictions=pyrogram.List([Restriction._parse(r) for r in user.restriction_reason]) or None,
+            photo=ChatPhoto._parse(
+                client, user.photo, user.id, user.access_hash),
+            restrictions=pyrogram.List(
+                [Restriction._parse(r) for r in user.restriction_reason]) or None,
```
[the remark this answers is not in the document]
Comment (tasi788) [reply pyrogram/client/types/user_and_chats/user.py]: I didn't notice that my IDE do autopep8 automatically, now should be good.

**Comment by delivrance** [comment]
This should be a property, by the way.

**Comment by delivrance** [comment]
I also experimented with custom formats in the past, which would allow things like `f"{user:mention}"` to be expanded into an HTML mention (this does actually exist in Pyrogram already, but is nowhere documented yet). Maybe we can have more, like `{u:full_name}` and such.

However, this only works for string interpolations and having `user.mention`, `user.full_name` and such properties is probably a better idea.

**Comment by JosXa** [comment]
I have a constant need for `display_name` (something that reliably gives you the most concrete, the most "clickable" version of a name):

```
def displayname(entity: Any) -> str:
    if hasattr(entity, "title"):
        return entity.title
    elif hasattr(entity, "first_name"):
        if entity.last_name:
            return f"{entity.first_name} {entity.last_name}"
        return entity.first_name
    raise ValueError("This entity does not seem to have a display name.")


def user_or_displayname(entity: Any) -> str:
    if entity.username:
        return f"@{entity.username}"
    return displayname(entity)
```

**Comment by JosXa** [comment]
Maybe also related?

```
from enum import IntEnum

from pyrogram import Client
from pyrogram.raw.types import Channel
from pyrogram.types import Message, User, Chat
from typing import Optional, Union, cast, Dict

_links_cache: Dict[int, str] = {}


class Platform(IntEnum):
    android = 1
    ios = 2
    web = 3
    desktop = 4


async def direct_link_to_message(
    reference: Message, platform: Optional[Platform] = Platform.android
) -> str:
    entity_link = await direct_link(reference._client, reference.chat, platform)
    return f"{entity_link}/{reference.message_id}"


async def direct_link(
    client: Client,
    peer: Union[User, Chat, Channel],
    platform: Optional[Platform] = Platform.android,
) -> str:
    if peer.username:
        return f"https://t.me/{peer.username}"
    else:
        if not isinstance(peer, User):
            peer_id = peer.id
            if isinstance(peer, Channel):
                return f"https://t.me/c/{peer_id}"
            invite_link = _links_cache.get(peer_id, None)
            if not invite_link:
                invite_link = (await client.get_chat(peer_id)).invite_link
                _links_cache[peer_id] = invite_link
            if invite_link:
                return cast(str, invite_link)

        if platform == Platform.android:
            return f"tg://openmessage?user_id={peer.id}"
        elif platform == Platform.ios:
            return f"t.me/@{peer.id}"
        elif platform == Platform.web:
            # TODO: maybe incorrect, test!
            return f"https://web.Telegram.org/#/im?p=u{peer.id}"
        else:
            raise ValueError(
                f"_IdentifiableUser has no username, creating direct link on platform {platform} not possible."
            )

```

**Pull request closed without merging**